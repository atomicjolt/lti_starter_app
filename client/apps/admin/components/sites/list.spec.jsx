import React from 'react';
import TestRenderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import List from './list';

const mockStore = configureStore([]);
const store = mockStore({
  settings: {
    canvas_callback_url: 'https://www.example.com',
  },
});

describe('sites list', () => {
  let result;
  let instance;

  const sites = {
    1: {
      id: 1,
      url: 'bfcoder.com'
    },

    2: {
      id: 2,
      url: 'atomicjolt.com'
    }
  };

  const deleteSite = () => {};

  beforeEach(() => {
    result = TestRenderer.create(
      <Provider store={store}>
        <List
          sites={sites}
          deleteSite={deleteSite}
        />
      </Provider>
    );
    instance = result.root;
  });

  it('matches the snapshot', () => {
    expect(result).toMatchSnapshot();
  });

  it('renders the list with header values', () => {
    const thead = instance.findByType('thead');
    const headings = thead.props.children.props.children;
    expect(headings.length).toEqual(3);
    expect(headings[0].type).toEqual('th');
    const spans = thead.findAllByType('span');
    expect(spans[0].props.children).toEqual('URL');
    expect(spans[1].props.children).toEqual('SETTINGS');
    expect(spans[2].props.children).toEqual('DELETE');
  });
});
