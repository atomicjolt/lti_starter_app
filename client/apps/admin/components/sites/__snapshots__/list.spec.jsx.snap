// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sites list matches the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <List
    sites={
      Object {
        "1": Object {
          "id": 1,
          "url": "bfcoder.com",
        },
        "2": Object {
          "id": 2,
          "url": "atomicjolt.com",
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <thead>
          <tr>
            <th>
              <span>
                URL
              </span>
            </th>
            <th>
              <span>
                SETTINGS
              </span>
            </th>
            <th>
              <span>
                DELETE
              </span>
            </th>
          </tr>
        </thead>,
        <tbody>
          <SiteRow
            site={
              Object {
                "id": 1,
                "url": "bfcoder.com",
              }
            }
          />
          <SiteRow
            site={
              Object {
                "id": 2,
                "url": "atomicjolt.com",
              }
            }
          />
        </tbody>,
      ],
      "className": "c-table c-table--lti",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <tr>
            <th>
              <span>
                URL
              </span>
            </th>
            <th>
              <span>
                SETTINGS
              </span>
            </th>
            <th>
              <span>
                DELETE
              </span>
            </th>
          </tr>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <th>
                <span>
                  URL
                </span>
              </th>,
              <th>
                <span>
                  SETTINGS
                </span>
              </th>,
              <th>
                <span>
                  DELETE
                </span>
              </th>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <span>
                  URL
                </span>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "URL",
                },
                "ref": null,
                "rendered": "URL",
                "type": "span",
              },
              "type": "th",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <span>
                  SETTINGS
                </span>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "SETTINGS",
                },
                "ref": null,
                "rendered": "SETTINGS",
                "type": "span",
              },
              "type": "th",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <span>
                  DELETE
                </span>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "DELETE",
                },
                "ref": null,
                "rendered": "DELETE",
                "type": "span",
              },
              "type": "th",
            },
          ],
          "type": "tr",
        },
        "type": "thead",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <SiteRow
              site={
                Object {
                  "id": 1,
                  "url": "bfcoder.com",
                }
              }
            />,
            <SiteRow
              site={
                Object {
                  "id": 2,
                  "url": "atomicjolt.com",
                }
              }
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "site_1",
            "nodeType": "class",
            "props": Object {
              "site": Object {
                "id": 1,
                "url": "bfcoder.com",
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "site_2",
            "nodeType": "class",
            "props": Object {
              "site": Object {
                "id": 2,
                "url": "atomicjolt.com",
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "tbody",
      },
    ],
    "type": "table",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <thead>
            <tr>
              <th>
                <span>
                  URL
                </span>
              </th>
              <th>
                <span>
                  SETTINGS
                </span>
              </th>
              <th>
                <span>
                  DELETE
                </span>
              </th>
            </tr>
          </thead>,
          <tbody>
            <SiteRow
              site={
                Object {
                  "id": 1,
                  "url": "bfcoder.com",
                }
              }
            />
            <SiteRow
              site={
                Object {
                  "id": 2,
                  "url": "atomicjolt.com",
                }
              }
            />
          </tbody>,
        ],
        "className": "c-table c-table--lti",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <tr>
              <th>
                <span>
                  URL
                </span>
              </th>
              <th>
                <span>
                  SETTINGS
                </span>
              </th>
              <th>
                <span>
                  DELETE
                </span>
              </th>
            </tr>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <th>
                  <span>
                    URL
                  </span>
                </th>,
                <th>
                  <span>
                    SETTINGS
                  </span>
                </th>,
                <th>
                  <span>
                    DELETE
                  </span>
                </th>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <span>
                    URL
                  </span>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "URL",
                  },
                  "ref": null,
                  "rendered": "URL",
                  "type": "span",
                },
                "type": "th",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <span>
                    SETTINGS
                  </span>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "SETTINGS",
                  },
                  "ref": null,
                  "rendered": "SETTINGS",
                  "type": "span",
                },
                "type": "th",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <span>
                    DELETE
                  </span>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "DELETE",
                  },
                  "ref": null,
                  "rendered": "DELETE",
                  "type": "span",
                },
                "type": "th",
              },
            ],
            "type": "tr",
          },
          "type": "thead",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <SiteRow
                site={
                  Object {
                    "id": 1,
                    "url": "bfcoder.com",
                  }
                }
              />,
              <SiteRow
                site={
                  Object {
                    "id": 2,
                    "url": "atomicjolt.com",
                  }
                }
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "site_1",
              "nodeType": "class",
              "props": Object {
                "site": Object {
                  "id": 1,
                  "url": "bfcoder.com",
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "site_2",
              "nodeType": "class",
              "props": Object {
                "site": Object {
                  "id": 2,
                  "url": "atomicjolt.com",
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "tbody",
        },
      ],
      "type": "table",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
